docs/source/tutorial/queries.mdx

```diff
@@ -121,132 +209,162 @@ const Launches: React.FC<LaunchesProps> = () => {
-            if (!fetchMoreResult) return prev;
-            return {
-              ...fetchMoreResult,
-              launches: {
-                ...fetchMoreResult.launches,
-                launches: [
-                  ...prev.launches.launches,
-                  ...fetchMoreResult.launches.launches,
-                ],
-              },
-            };
-          },
-        })
-      }
-    >
-      Load More
-    </Button>
-  )
-}
+{data.launches && data.launches.hasMore && (
+  isLoadingMore
+    ? <Loading />
+    : <Button
+        onClick={async () => {
+          setIsLoadingMore(true);
+          await fetchMore({
+            variables: {
+              after: data.launches.cursor,
+            },
+          });
+          setIsLoadingMore(false);
+        }}
+      >
+        Load More
+      </Button>
+)}
+//</Fragment>
 ```
 
-```jsx:title=src/pages/launches.jsx
-{data.launches &&
-  data.launches.hasMore && (
-    <Button
-      onClick={() =>
-        fetchMore({ // highlight-line
-          variables: {
-            after: data.launches.cursor,
-          },
-          updateQuery: (prev, { fetchMoreResult, ...rest }) => { // highlight-line
-            if (!fetchMoreResult) return prev;
+</MultiCodeBlock>
+
+When our new button is clicked, it calls `fetchMore` (passing the current `cursor` as the value of the `after` variable) and displays a Loading notice until the query returns results.
+
+Let's test our button. Start everything up and visit `localhost:3000` again. A **Load More** button now appears below our 20 launches. Click it. After the query returns, _no additional launches appear_. 🤔
+
+If you check your browser's network activity, you'll see that the button did in fact send a followup query to the server, and the server did in fact respond with a list of launches. _However_, Apollo Client keeps these lists separate, because they represent the results of queries with _different variable values_ (in this case, the value of `after`).
+
+We need Apollo Client to instead _merge_ the launches from our `fetchMore` query with the launches from our _original_ query. Let's configure that behavior.
+
+### Merge cached results
+
+Apollo Client stores your query results in its **in-memory cache**. The cache handles most operations intelligently and efficiently, but it doesn't automatically know that we want to merge our two distinct lists of launches. To fix this, we'll define a **`merge` function** for the paginated field in our schema.
+
+Open `src/cache.ts`, where our default `InMemoryCache` is initialized:
+
+<MultiCodeBlock>
+
+```ts:title=src/cache.ts
+import { InMemoryCache, Reference } from '@apollo/client';
+
+export const cache: InMemoryCache = new InMemoryCache({});
+```
+
+</MultiCodeBlock>
+
+The schema field that our server paginates is the list of `launches`. Let's add a `merge` function for this field to the cache's configuration, like so:
```

Comment:
This seems like a good place to link to the new pagination documentation I've been working on in https://github.com/apollographql/apollo-client/pull/7175.

Reply:
I have an open item to do this as soon as that doc is live (I'll review it tomorrow!)